skimage/util/lazy.py

```diff
@@ -0,0 +1,65 @@
+import sys
+import importlib
+import importlib.util
+
+
+def install_lazy(module_name, submodules=None, submod_funcs=None):
+    """Install lazily loaded submodules and functions.
+
+    Parameters
+    ----------
+    module_name : str
+        Typically use __name__.
+    submodules : list
+        List of submodules to install.
+    submod_funcs : dict
+        Dictionary of submodule -> list of functions.  These functions are
+        imported as they are used.
+    """
+    if submod_funcs is None:
+        submod_funcs = {}
+
+    if submodules is None:
+        submodules = []
+
+    all_funcs = []
+    for mod, funcs in submod_funcs.items():
+        all_funcs.extend(funcs)
+
+    def __getattr__(name):
+        if name in submodules:
```

Comment:
Cast `submodules` to a `set` before doing this for an `O(1)` test instead of an `O(N)` test.

Reply:
Done